TeamProject/Engine/Engine/Temperature.cs

```diff
@@ -0,0 +1,55 @@
+﻿
+using System;
+
+namespace Engine
+{
+    /// <summary>
+    /// Wrapper class that puts limits on the temperature
+    /// </summary>
+    public class Temperature
+    {
+        private double value;
+
+        public double Value
+        {
+            get => value;
+            set => SetTemp(value);
+        }
+
+        public Temperature(double t) {
+            SetTemp(t);
+        }
+
+        public void SetTemp(double temp)
```

Comment:
You can simplify this by using `Math.Clamp` 
see https://docs.microsoft.com/en-us/dotnet/api/system.math.clamp?view=netcore-3.0

Reply:
also you might want to use `Math.Round` see https://docs.microsoft.com/en-us/dotnet/api/system.math.round?view=netframework-4.8